packages/wrangler/src/deployment-bundle/entry.ts

```diff
@@ -112,6 +113,9 @@ export async function getEntry(
 		directory,
 		format,
 		moduleRoot: args.moduleRoot ?? config.base_dir ?? path.dirname(file),
+		staticAssetsOnly:
+			!!(args.experimentalAssets || config.experimental_assets) &&
+			!(args.script && config.main),
```

Comment:
In `packages/wrangler/src/deployment-bundle/create-worker-upload-form.ts`, you have a check for if `staticAssetsOnly`, and if yes, only ever send the assets piece. Fair enough!

But here, you're also saying that `staticAssetsOnly` is not true if any user worker script is set.

I think you either want to loosen this to say that if there is experimentalAssets/experimental_assets is defined, then that just always means `staticAssetsOnly` (forget about the presence of script/main).

```suggestion
```

*OR* you want to just always try attaching the `assets` piece in metadata and let the API fail if it fails. We'll get there eventually, but I have a slight personal preference of the former, just to reduce the work you need to do and keep it consistent with all the tests you've already written :)

Reply:
On that note, might be nice to have another test which *does* directly reference a user Worker (e.g. with `main` config) and to see how that collapses.

Reply:
I might be misunderstanding your comment, but the idea _was_ to have `staticAssetsOnly` as a flag for when `experimental-assets` was used without a user worker. This was to make it easier to pass through to `create-worker-upload-form` to make sure the `no-op-asset-worker` wasn't uploaded. Admittedly this was a bit of a janky temp solution, since I think this wont be necessary once we can get rid of the `no-op-asset-worker` that `wrangler dev` needs. 

Upon reflection I think `Entry` does seem an inappropriate place for this, so I've moved it to the `experimentalAssets` options that gets passed through to `create-worker-upload-form`,

Reply:
Yeah, tad janky but you've improved it a bit and like you say, it's temporary anyway and we'll be in amongst here plenty in future PRs for user worker pieces when they're etc., so won't block!

Reply:
Is there still a no-op asset worker passed to `wrangler dev` as an entry?
It feels like this is no longer needed and we could just say that the entry is undefined when there are only assets?

Reply:
Oh I see that there is a `no-op-asset-worker` in the templates. Do we really need this? Isn't the work that Carmen has done avoid the need for this?